File: cpp/searching/linearSearch.cpp
# include <iostream>
using namespace std;

// BIG - O
// o(n)

int linearSearch(int arr[], int size ,int value) {
  for (int i = 0; i < size; i++) {
    if (arr[i] == value) {
      return i;
    }
  }
  return -1;
}

int main() {
  int arr[]= {3, 5 ,22, 12, 43, 345, 65, 23};
  int size = sizeof(arr) / sizeof(arr[0]);

  cout << linearSearch(arr, size, 34) << endl;
  cout << linearSearch(arr, size, 345) << endl;
  return 0;
}

File: js/searching/linearSearch.js
// Using for loop
const linearSearchForLoop = function (arr, val) {
  for (let i = 0; i < arr.length; i++) {
    if (arr[i] === val) return i;
  }
  return -1;
};

// Using recursion
const linearSearch = function (arr, val) {
  if (arr.length === 0) return false;
  if (arr[0] === val) return true;
  return linearSearch(arr.slice(1, arr.length), val);
};

const arr = [1, 3, 1, 2, 0, 19, 23];
console.log(linearSearch(arr, 233));
